src/test/java/space/obminyashka/items_exchange/controller/AdvertisementControllerTest.java

```diff
@@ -102,4 +107,123 @@ void findPaginatedAsThumbnails_shouldReturnProperQuantityOfAdvertisementsThumbna
                 .andExpect(status().isOk())
                .andExpect(jsonPath("$.length()").value(advertisementRepository.findAll().size()));
     }
+
+    @Test
+    @WithMockUser(username = "admin")
+    void updateAdvertisement_shouldReturn400WhenNotValidAdvertisementSize() throws Exception {
+        AdvertisementDto existDtoForUpdate = AdvertisementDtoCreatingUtil
+                .createExistAdvertisementDtoForUpdateWithNotValidSize();
+
+        MvcResult mvcResult = mockMvc.perform(put("/adv")
+                .content(asJsonString(existDtoForUpdate))
+                .contentType(MediaType.APPLICATION_JSON)
+                .accept(MediaType.APPLICATION_JSON))
+                .andDo(print())
+                .andExpect(status().isBadRequest())
+                .andReturn();
+
+        assertTrue(mvcResult
+                .getResponse()
+                .getContentAsString()
+                .contains("must be between 1 and 50 symbols"));
+    }
+
+    @Test
+    @WithMockUser(username = "admin")
+    void updateAdvertisement_shouldReturn400WhenNotValidAdvertisementLocation() throws Exception {
+        AdvertisementDto existDtoForUpdate = AdvertisementDtoCreatingUtil
+                .createExistAdvertisementDtoForUpdateWithNotValidLocation();
+
+        String expectedResponseArea =
+                "area: " + existDtoForUpdate.getLocation().getArea() + " must be between 2 and 100 symbols";
+        String expectedResponseDistrict =
+                "district: " + existDtoForUpdate.getLocation().getDistrict() + " must be between 2 and 100 symbols";
+        String expectedResponseCity =
+                "city: " + existDtoForUpdate.getLocation().getCity() + " must be between 2 and 100 symbols";
+
+        MvcResult mvcResult = mockMvc.perform(put("/adv")
+                .content(asJsonString(existDtoForUpdate))
+                .contentType(MediaType.APPLICATION_JSON)
+                .accept(MediaType.APPLICATION_JSON))
+                .andDo(print())
+                .andExpect(status().isBadRequest())
+                .andReturn();
+
+        Assertions.assertAll(
+                () -> assertTrue(mvcResult
+                .getResponse()
+                .getContentAsString()
+                .contains(expectedResponseArea)),
+
+                () -> assertTrue(mvcResult
+                .getResponse()
+                .getContentAsString()
+                .contains(expectedResponseDistrict)),
+
+                () -> assertTrue(mvcResult
+                .getResponse()
+                .getContentAsString()
+                .contains(expectedResponseCity))
+        );
+    }
+
+    @Test
+    @WithMockUser(username = "admin")
+    void updateAdvertisement_shouldReturn400WhenNotValidAdvertisementTopic() throws Exception {
```

Comment:
Why don't use one test for all possible validation errors?

Reply:
ok, i`ll create one test